doc: 54H custom pcb guide update (#16796)
nrfconnect/sdk-nrf: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_custom_pcb.rst

Author: @tjora Thanks for the feedback, implemented your suggestions. Please re-review

First, you need to create your PCB for the nRF54H20 SoC.

We highly recommend using the PCB layouts and component values provided by Nordic Semiconductor, especially for clock and power sources, considering the following limitations:

* The DC/DC inductor must be present on the PCB.

> * the DC/DC power converter must be present on the PCB and enabled.

Reviewer: What is the DC/DC power converter?

Author: @tjora Can you verify this?

Reviewer 2: I think the first two bullet points could be merged like this:
```
* The DC/DC inductor must be present on the PCB and the ``inductor-present;`` node node must be configured either in :file:`zephyr/boards/your_board/nrf54h20dk_bicr.dtsi` or where the section ``bicr: bicr@fff87b0`` is present in the devicetree.
```

Reviewer 3: @ingeryp suggestion is great and to the point, please use that one.

  Also, the ``inductor-present;`` node must be configured either in :file:`zephyr/boards/your_board/nrf54h20dk_bicr.dtsi` or where the section ``bicr: bicr@fff87b0`` is present in the devicetree.
* For the lowest sleep power consumption, use a 32 KHz crystal.
* The **P9** port cannot be used with internal nor external pull-down resistors.
* For optimal performance, the output impedance of the **P6** and **P7** ports should match the PCB and external device pin impedance.

>   However, an existing limitation requires this configuration to be 50 ohms.

Author: Updated @David-Garcia-Polo

First, you need to create your PCB for the nRF54H20 SoC.

We highly recommend using the PCB layouts and component values provided by Nordic Semiconductor, especially for clock and power sources, considering the following limitations:

* The DC/DC inductor must be present on the PCB.
* the DC/DC power converter must be present on the PCB and enabled.

>   Also, the ``inductor-present;`` node must be configured either in :file:`zephyr/boards/your_board/nrf54h20dk_bicr.dtsi` or where the section ``bicr: bicr@fff87b0`` is present in the devicetree.

Reviewer 2: I am not sure it makes sense for the customer to have any file prefixed with `nrf54h20dk` in their board definition.

Reviewer 3: Good point, suggest to drop the reference to the file completely, something like
Also, the ``inductor-present;`` node must be configured in the section ``bicr: bicr@fff87b0`` in the devicetree.

I guess it somewhere is described how to make your own board specific devicetree, maybe that could be referenced?

* For the lowest sleep power consumption, use a 32 KHz crystal.
* The **P9** port cannot be used with internal nor external pull-down resistors.
* For optimal performance, the output impedance of the **P6** and **P7** ports should match the PCB and external device pin impedance.

Prepare the configuration files for your custom board in the |NCS|
******************************************************************

##TODO

The nRF54H20 DK uses multiple board files for its configuration:

> * `ARM cores configuration files`_

Reviewer 2: These links point to `next` repos, and should be removed.

I believe the copying you are supposed to do now is:
`sdk-zephyr/boards/nordic/nrf54h20dk` -> `sdk-zephyr/boards/<customer>/<product>`

* `Risc-V cores configuration files`_

You can use these files as a starting point for configuring your own custom board.
The easiest way to do so, when creating a :ref:`Zephyr repository application <zephyr:zephyr-repo-app>`, is to create a copy of these folders under :file:`sdk-zephyr/boards/your_vendor_prefix/your_custom_board_name`.

.. caution::

>    Do not modify the configuration files related to the Secure Domain (:file:`*_cpusec` in the ARM folder) and the System Controller (:file:`*_cpusys` in the Risc-V folder).

Reviewer 2: I cannot see any configuration files related to the Secure Domain or the System controller in `sdk-zephyr/boards/nordic/nrf54h20dk`. I believe these are now kept in `sdk-secdom` and `sdk-sysctrl` respectively, so they should not interfere here.

Reviewer 2: Also the `nRF54H20_common_SoC_files` link is no longer valid. I think that should be pointing to `sdk-zephyr/dts/common/nordic/`

Reviewer 2: I think you should definitely add a reference to this page: https://docs.nordicsemi.com/bundle/ncs-latest/page/zephyr/hardware/porting/board_porting.html

First, you need to create your PCB for the nRF54H20 SoC.

We highly recommend using the PCB layouts and component values provided by Nordic Semiconductor, especially for clock and power sources, considering the following limitations:

* The DC/DC inductor must be present on the PCB, and the ``inductor-present;`` node must be configured either in the :file:`zephyr/boards/your_board/nrf54h20dk_bicr.dtsi` file or in the section where ``bicr: bicr@fff87b0`` is present in the devicetree.
* For the lowest sleep power consumption, use a 32 KHz crystal.

> * The **P9** port cannot be used with internal nor external pull-down resistors.

Reviewer 4: ```suggestion
* The **P9** port cannot be used with internal or external pull-down resistors.
```

* For optimal performance, the output impedance of the **P6** and **P7** ports should match the PCB and external device pin impedance.
* Use one of the following power supply options:

>   * VDDH higher than 2.2V.
>   * VDDH shorted to VDD at 1.8V.

Reviewer 4: ```suggestion
  * VDDH higher than 2.2V
  * VDDH shorted to VDD at 1.8V
```
since these are not sentences

Prepare the configuration files for your custom board in the |NCS|
******************************************************************

The nRF54H20 DK uses multiple board files for its configuration.
You can use these files as a starting point for configuring your own custom board.

> The easiest way to do so, when creating a :ref:`Zephyr repository application <zephyr:zephyr-repo-app>`, is to create a copy of files in :file:`sdk-zephyr/boards/nordic/nrf54h20dk` under :file:`sdk-zephyr/boards/<your_vendor_prefix>/<your_custom_board_name>`.

Reviewer 4: ```suggestion
When creating a :ref:`Zephyr repository application <zephyr:zephyr-repo-app>`, copy the files from :file:`sdk-zephyr/boards/nordic/nrf54h20dk` to the :file:`sdk-zephyr/boards/<your_vendor_prefix>/<your_custom_board_name>` file.
```

* `Risc-V cores configuration files`_

You can use these files as a starting point for configuring your own custom board.

> The easiest way to do so, when creating a :ref:`Zephyr repository application <zephyr:zephyr-repo-app>`, is to create a copy of these folders under :file:`sdk-zephyr/boards/nordic/your_custom_board_name`.

Reviewer 5: ```suggestion
The easiest way to do so, when creating a :ref:`Zephyr repository application <zephyr:zephyr-repo-app>`, is to create a copy of these folders under :file:`sdk-zephyr/boards/your_vendor_prefix/your_custom_board_name`.
```

Reviewer 6: Shouldn't this be conformant with HWMv2 and link to https://docs.nordicsemi.com/bundle/ncs-latest/page/zephyr/hardware/porting/board_porting.html#board-terminology or https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/app_dev/board_support/board_names.html ?

Author: Is the board target supposed to match the board's folder structure? @gmarull @greg-fer

> The easiest way to do so, when creating a :ref:`Zephyr repository application <zephyr:zephyr-repo-app>`, is to create a copy of these folders under :file:`boards/arm/your_custom_board_name` (for the ARM configuration files) and :file:`boards/riscv/your_custom_board_name` (for the Risc-V configuration files), respectively.

Reviewer 5: `boards/arch` no longer exist, it's now `boards/vendor`.

Author: @gmarull updated

Author: @gmarull updated

* Tamper switches
* Active shield channels

You can find the details in the DTS specification for the BICR in :file:`sdk-zephyr/dts/bindings/misc/nordic,nrf-bicr.yaml`

> When not set, the registers' default value is ``0xFFFFFFFF``.

Reviewer 7: From above:
"The Board Information Configuration Registers (BICR) are non-volatile memory (NVM) registers"
And here, we talk about a single register. Is this one or more?

Author: This is out of scope of this guide. Registers should be part of a separate PDF document.
If it is confusing, I can remove this reference completely @hellesvik-nordic

* Tamper switches
* Active shield channels

> You can find the details of each register contained in BICR in the relevant `BICR register's PDF file`_.

Reviewer 2: With this reference gone I still think we might need to point to some other entry, otherwise the dts entries themselves can seem a bit magical.

I guess we could either link to the PS, or we can link to the dts specification for the bicr: `sdk-zephyr/dts/bindings/misc/nordic,nrf-bicr.yaml`

Author: I guess we will have to use the yaml file then, as the PS is still in the works.

BICR configuration
==================

The nRF54H20 DK BICR configuration can be found in the board configuration directory as :file:`sdk-zephyr/boards/nordic/nrf54h20dk/nrf54h20dk_bicr.dtsi`.
This file is used by the |NCS| build system to generate a corresponding HEX file.
You can start from this file when editing the values of the devicetree properties inside your custom board folder (:file:`boards/nordic/your_custom_board`), according to your board configuration.

.. caution::

>    A mismatch between the board and the configuration values in BICR can damage the device or set it in an unrecoverable state.

Reviewer 7: Are there any specific things that will cause this that we should point out? Are for example a low frequency clock or power configurations the ones that can do so, while others will likely not put it into an unrecoverable state?
If so, maybe we can point out something users can keep an extra eye on?

Author: @David-Garcia-Polo @ingeryp @tjora Any inputs here?

BICR configuration
==================

##TODO

> The nRF54H20 dk BICR configuration can be found in the board configuration directory as :file:`boards/arm/nrf54h20dk_nrf54h20/nrf54h20soc1_pdk_bicr.dtsi`.

Reviewer 8: boards/arm/nrf54h20dk_nrf54h20/nrf54h20soc1_pdk_bicr.dtsi`. this was for SOC1 
now it is alreadu upstreamed to:
zephyr/boards/nordic/nrf54h20dk/nrf54h20dk_bicr.dtsi

This file is used by the |NCS| build system to generate a corresponding HEX file.
You can start from this file when editing the values of the devicetree properties inside your custom board folder (:file:`boards/arm/your_custom_board`), according to your board configuration.

Generating the BICR binary
==========================

##TODO

> To generate the BICR binary, you must first set the Kconfig option :kconfig:option:`CONFIG_INCLUDE_BICR` to ``y``.

Reviewer 8: CONFIG_INCLUDE_BICR -> CONFIG_NRF_REGTOOL_GENERATE_BICR
see https://github.com/nrfconnect/sdk-zephyr/blob/95fbf6c2ef3ae98a3f5243229b5d20a37dcb984a/soc/nordic/nrf54h/Kconfig.defconfig.nrf54h20_cpuapp

Author: @lstnl  updated

BICR configuration
==================

##TODO

The nRF54H20 dk BICR configuration can be found in the board configuration directory as :file:`sdk-zephyr/boards/nordic/nrf54h20dk/nrf54h20dk_bicr.dtsi`.
This file is used by the |NCS| build system to generate a corresponding HEX file.
You can start from this file when editing the values of the devicetree properties inside your custom board folder (:file:`boards/nordic/your_custom_board`), according to your board configuration.

> 

Reviewer: Can you add a big note here saying that a mismatch between the board and the configuration values in BICR may damage the device or set it in an unrecoverable state?

Author: @David-Garcia-Polo done

After the |NCS| build system generates the BICR binary, you must flash this binary manually.
The content of BICR should be loaded to the SoC only once and should not be erased nor modified unless the PCB layout changes.
To manually program the generated :file:`bicr.hex` file to the SoC, use nRF Util as follows::

    nrfutil device program --options chip_erase_mode=ERASE_NONE --firmware bicr.hex --core Application --serial-number <serial_number>

You only need to follow this programming process once, assuming the PCB configuration applied through the BICR is correct the first time.
However, it is also possible to reprogram the BICR while in the LCS ``RoT``.
This can be useful, for example, when adjusting the configuration as the PCB design gets refined or modified, requiring the process to be repeated.

Validate the BICR binary
------------------------

> After programming the BICR binary onto the device, you should validate whether the BICR works with your device.

Reviewer 4: ```suggestion
After programming the BICR binary onto the device, validate whether the BICR works with your device.
```

> 1. Reset the device::

Reviewer 4: ```suggestion
1. Reset the device by running the following command::
```

Flashing the BICR binary
========================

##TODO

After the |NCS| build system generates the BICR binary, you must flash this binary manually.
The content of BICR should be loaded to the SoC only once and should not be erased nor modified unless the PCB layout changes.
To manually program the generated :file:`bicr.hex` file to the SoC, use nRF Util as follows::

    nrfutil device program --options chip_erase_mode=ERASE_NONE --firmware bicr.hex --core Application --serial-number <serial_number>

You only need to follow this programming process once, assuming the PCB configuration applied through the BICR is correct the first time.
However, it is also possible to reprogram the BICR while in the LCS ``RoT``.
This can be useful, for example, when adjusting the configuration as the PCB design gets refined or modified, requiring the process to be repeated.

Validate the BICR binary
------------------------

After programming the BICR binary onto the device, you should validate whether the BICR was successfully installed and executed.

> Reset the device

Reviewer 7: If I program the BICR and then go to this step, then I want to validate it.
Then I come to this step, and I see "ah okay I reset the device".
Now if my BICR is wrong, and the device is not operational, then I run the chance that I cannot recover it right?
Would it not be better to describe the nrfutil method before the reset method, because the nrfutil method below assumedly does run the risc or breaking the device?

Reviewer 7: Since the device needs to be reset to do the seconds step, I close this comment.
See https://github.com/nrfconnect/sdk-nrf/pull/16796#discussion_r1745497275

Reviewer 2: Maybe I can explain a bit better.

We have no way to test the BICR, except for programming and loading it through a reset, then seeing what happens.

So the procedure is:
1. We program the BICR
2. We reset the chip so that the System Controller ROM code can apply the power settings
3. We double check that the System Controller ROM successfully applied the power settings

So the actual customer procedure I would suggest is:
1. Program the BICR
2. Reset the chip
3. `nrfutil device x-read --address 0x2F88FF1C --serial-number 1051164514 --core Secure`
We then look at what happens during step 3:
 - Error: Device is likely trash :(
 - `0xCE68C97C` (BICR loading failed): You _should_ be able to program another BICR
 - `0xD78213DF` (No BICR present): Customer did not actually program a BICR, try again :)
 - `0x289CFB73` (Success): BICR settings were applied and system appears to behave nominally. Proceed with bringup

Note:
This will _only_ validate the parts of the BICR relating to power as the clock section is not applied by ROM code, but rather by System Controller FW, starting in ROT. However, the clock part being incorrect _should_ still allow you to replace the BICR.

The reason we should at least in part validate the BICR here, is that if we do not, then the failure may not become apparent until after the LCS transition to ROT, in which case it gets very hard to prove that the issue is with the BICR, and not with SDSC, or some other part of the system. We can also only read this ROM report in Empty, so its a good idea to do it here.

Reviewer 2: I guess sysrom might also apply tamper controller and shield settings, @David-Garcia-Polo would have to answer that, as I have never played around with those things.

   The simplest way to check if the BICR is functioning is to reset the device.

>    If the device is still operational after the reset, the BICR is at least partially loaded.

Reviewer 7: When I read this as a user, the first thing I think is: "... and what if it is not operational?"

Author: @hellesvik-nordic Sure, that makes sense. Could you describe how to detect that?

Reviewer 7: I do not really know how to detect that. I have not written wrong BICR to a board yet.
Is there maybe a specific nrfutil error message that would pop up if BICR is broken? 
Maybe something along the lines of "If after a reset, nrfutil gives the error [INSERT ERROR], meaning that Jlink cannot find the device, then the device is broken because the BICR was wrong.".
I too hope Inge can help getting this right.

Reviewer 2: Yeah, the problem is that if the BICR is invalid then the chip can be in an undefined state, so almost anything can happen, and almost any error can be returned. It would of-course depend on the specifics of the board, and the BICR. I dont remember the exact messages offhand, but this is a list roughly in order, which would all have separate errors:

* JLink error due to bad voltage on the SWD port
* DAP fails to power up - probably the most likely one, I have seen these on kits more than once
* Failed to identify the device - indicating other power issues in the DAP
* Access ports fail to open - This indicates that the SDROM crashed before it was able to open the Secure Domain access port
* Memory access errors when accessing peripherals
* MRAM read/write errors

Reviewer: 1) Is the inductor really needed? YES. It is needed and not optional.
2) Why does BICR.POWER.CONFIG:INDUCTOR not make any sense to you? You must place the inductor in the board and you must write in the BICR that the inductor is present.

Reviewer 7: Since proving the negative here will be hard, I think trying to describe it can cause confusion.
Therefore I will close this conversation, but we can keep it in mind for later to maybe improve this when we got more experience with users trying to follow the guide

>    If the device is still operational after the reset, the BICR is at least partially loaded.

Reviewer: The device will be operational if the BICR is obviously wrong, so I do not think it is a good indication. And the problem is that a lot of BICRs are valid and correct but not supported. right now we only support 2 different power supplies option: (1) VDDH > 2.2V and inductor present and (2) VDDH shorted to VDD at 1.8V and inductor present. They should choose one of these two, make the board like this and choose one of two BICRs. 

A simple check could be done by seeing if it is one of these two.

Author: I removed this reference and added your power supply suggestions in the prepare your pcb section

Validate using nrfutil

>   When in LCS ``EMPTY``, you can use `nrfutil` to explicitly verify the BICR status by reading the memory::

Reviewer 7: I don't know this, so i am fumbling a bit in the dark, but:
Must the device be reset before x-read can be done, or can it be done right after loading the BICR, no reset needed in between?
Im asking cause one of the errors says "no found during boot", and that seems to me it would require a reboot before running the check?

Reviewer: Actually the address is populated at every step. So there will be a value there if there is not reset. They need to reset. But I  do not know if this is the best way to check.

     nrfutil device x-read --address 0x2F88FF1C --serial-number 1051164514 --core Secure

  nrfutil returns the BICR loading status using one of the following values:

  * ``0x289CFB73``: BICR loaded without error.
  * ``0xD78213DF``: No BICR was loaded (as none was found during the boot).

>   * ``0xCE68C97C``: BICR loading failed (device can be unresponsive and not being able to print this value).

Reviewer 7: If a user gets any of these error codes, we should add some instructions on what to do then.

Programming the SDFW and SCFW
=============================

> ##TODO

Reviewer 2: The HW validates the checksum on the sdsc bundle when you try to transition to LCS ROT.
If something went wrong during programming of the sdsc bundle, then you will get an error on `nrfutil device x-adac-lcs-change` saying `ADAC_FAILURE`

> ##TODO

Author: in case of success, what would be the output of that command? @ingeryp

> ##TODO

Reviewer 2: ```
$ nrfutil device x-adac-lcs-change --life-cycle rot --serial-number 001051125716 
✔️ Changed life cycle state 1051125716
```

Flashing the BICR binary
========================

##TODO

After the |NCS| build system generates the BICR binary, you must flash this binary manually.
The content of BICR should be loaded to the SoC only once and should not be erased nor modified unless the PCB layout changes.

> To manually flash the generated :file:`bicr.hex` file to the SoC, use nRF Util as follows::

Reviewer 8: bicr.hex can be programmed like this
`nrfutil device program --options chip_erase_mode=ERASE_NONE --firmware bicr.hex --core Application --serial-number <DK_SERIAL_NUMBER>`

but I'm not sure if west flashwill not do it also.

Author: Is the --core parameter supposed to be ``application`` instead of ``secure``?

Reviewer 8: Yes secure can be used only for secdom and when LCS is empty. In ROT You will get error when trying to use --core secure

Reviewer 2: > but I'm not sure if west flashwill not do it also.
I dont think west flash should program anything that is likely to brick your device. I think this should be kept nrfutil only.

Flashing the BICR binary
========================

##TODO

After the |NCS| build system generates the BICR binary, you must flash this binary manually.
The content of BICR should be loaded to the SoC only once and should not be erased nor modified unless the PCB layout changes.
To manually flash the generated :file:`bicr.hex` file to the SoC, use nRF Util as follows::

    nrfutil device program --options chip_erase_mode=ERASE_NONE --firmware bicr.hex` --core Application --serial-number <serial_number>

> You need to follow this flashing process only one time, as the PCB configuration will not change.

Author: What's the command to program the BICR in RoT? Is it any different from the previous one? @tjora

> You need to follow this flashing process only one time, as the PCB configuration will not change.

Reviewer 2: The command would be identical, but you are using `--core Application` rather than `--core Secure`.

Note that I do not think we have done much testing on how to perform this replacement in ROT, so I am not sure its something we should recommend in the docs.

@David-Garcia-Polo suggested something like this, as the safe option (1):
1. Erase the existing BICR
2. Modify the board
3. Program the new BICR
4. Reset

Another one that would probably work (2):
1. Program the new BICR
2. Power of the device _before any resets_
3. Modify the board
4. Reset

If you do not need to modify the board, and just need to patch some values then this would be fine(3):
1. Program the new BICR
2. Reset

Strategy (1) would require that we test that sdsc will allow us to program a BICR if no BICR is programmed.
Strategy (2) would require that we have a lot of confidence that that are no undue resets between programming and powering of the board. Personally I think this is quite scary.
Strategy (3) should be fine, but I do not know how well this is tested by SysFW devs.

> 

Reviewer 2: I think you can probably add another section here to validate the BICR binary at this stage already.

Currently the "best" way to do this is just to reset this device and see if it still works.
There are some additional validation steps that we could take (and potentially integrate into the tools if needed), that will warn about some classes of errors. However, as many BICR errors are deadly to the HW, the fact you can still work with the device afterwards indicates that the BICR is at least partially okay.

Reviewer 2: One slightly ugly way to validate that the BICR is installed and was successfully run is to do this:
```
(venv) inry@inry-linux:~/proto/razer_bicr$ nrfutil device x-read --address 0x2F88FF1C --serial-num
ber 1051164514 --core Secure
1051164514
0x2F88FF1C: 289CFB73                              |s..(|
```
Values can be decoded like this:
```
        0x289CFB73: BICR was loaded without error.
        0xCE68C97C: There was a failure loading the BICR. Note: you would typically not see this value, as the chip would be dead...
        0xD78213DF: No BICR was loaded (as none could be found during bootup).
```
This can only be done in LCS Empty

Reviewer 2: I would suggest more along these lines:

After programming the BICR binary onto the device, you should ensure that the BICR works with your device.
To do this perform the following steps:
* `nrfutil device reset --reset-kind RESET_PIN --serial-number <serial_number>`
* `nrfutil device x-read --address 0x2F88FF1C --serial-number 1051164514 --core Secure`

The second command has 4 plausible outcomes:
* Error - This indicates that the device is likely suffering from severe power issues after applying the BICR. This state is likely unrecoverable.
* ``0xCE68C97C``: BICR application failed. This indicates that there is an issue with the BICR, but in most cases this can be recovered by programming the correct BICR.
* ``0xD78213DF``: BICR application was skipped. Indicating that no BICR was programmed to the device.
* ``0x289CFB73``: BICR applied without error. This indicates that the power configuration of the BICR is valid and you can proceed to the next step.

Programming the SDFW and SCFW
=============================

##TODO

After programming the BICR, the nRF54H20 SoC requires the provisioning of a bundle ( :file:`nrf54h20_soc_binaries_v0.5.0.zip`) containing the precompiled firmware for the Secure Domain and System Controller.
To program the Secure Domain Firmware (SDFW, also known as ``urot``) and the System Controller Firmware (SCFW) from the firmware bundle to the nRF54H20 DK, do the following:

1. Download the `nRF54H20 firmware bundle v0.5.0`_.
#. Move the :file:`ZIP` bundle to a folder of your choice.
#. |open_terminal_window_with_environment|

> #. Run nRF Util to program the binaries using the following command::

Author: Is there an easy way to verify the current LCS? @tjora

> #. Run nRF Util to program the binaries using the following command::

Author: Would this command work in both empty and rot? @tjora nrfutil device x-suit-dfu --firmware <path-to_bundle_zip_file> --serial-number <serialnumber>

> #. Run nRF Util to program the binaries using the following command::

Reviewer 2: The command would be identical, but you are using `--core Application` rather than `--core Secure`.

Note that I do not think we have done much testing on how to perform this replacement in ROT, so I am not sure its something we should recommend in the docs.

@David-Garcia-Polo suggested something like this, as the safe option (1):
1. Erase the existing BICR
2. Modify the board
3. Program the new BICR
4. Reset

Another one that would probably work (2):
1. Program the new BICR
2. Power of the device _before any resets_
3. Modify the board
4. Reset

If you do not need to modify the board, and just need to patch some values then this would be fine(3):
1. Program the new BICR
2. Reset

Strategy (1) would require that we test that sdsc will allow us to reprogram the chip if BICR is not programmed.
Strategy (2) would require that we have a lot of confidence that that are no undue resets between programming and powering of the board. Personally I think this is quite scary.
Strategy (3) should be fine, but I do not know how well this is tested by SysFW devs.

> #. Run nRF Util to program the binaries using the following command::

Reviewer 2: The `x-suit-dfu` command will only work in ROT @FrancescoSer , as the DFU procedure is performed by talking to present and running sdfw, which only happens in ROT.

> #. Run nRF Util to program the binaries using the following command::

Author: Should these steps also assume that users might want to upgrade the bundle on an existing custom board? Also is there a scenario where the user might want to update also the BICR? (if yes, how?) @ingeryp @tjora @David-Garcia-Polo

> #. Run nRF Util to program the binaries using the following command::

Reviewer 2: ## Update BICR:
Yes, customer may want to do this if they make modifications to their PCB, such as supplying an external square wave rather than the on-board oscillator, to tweak certain values, or to change the power supply mode.

You can reprogram the BICR in LCS ROT+ using this command:
`nrfutil device program --firmware <bicr.hex> --core Application --serial-number <snr>`
However the exact sequencing is not clear, and most importantly _not tested_.
See this comment https://github.com/nrfconnect/sdk-nrf/pull/16796#discussion_r1745303511

## Update SDSC
Yes, I would assume customers would need to do this. This can be done with the following commands
```
unzip nrf54h20_soc_binaries_v<x.y.z>.zip
nrfutil device x-suit-dfu --serial-number $snr --firmware nordic_top.suit
```
You can verify the upgrade by running this command before and after you upgrade sdsc. The version number should increase after the dfu procecure is complete:
```
nrfutil device x-sdfw-version-get --firmware-slot uslot --serial-number <snr>
```
@hakonfam could expand on this if needed.

> #. Run nRF Util to program the binaries using the following command::

Author: If the BICR update sequence is not tested, I'll not document it right now

> #. Run nRF Util to program the binaries using the following command::

Author: Added the update section @ingeryp

* When creating a new application specific to your new board, DTS board files can contain all necessary configurations, and no overlay file is needed.
  However, the same limitations regarding the consistency and UICR configuration apply, but should be kept on the board files level.

* You must manually program the BICR if it has been modified.

Update the SDFW and SCFW
************************

> When a new version of the nRF54H20 firmware bundle is released, you can proceed with the update as follows:

Reviewer 4: ```suggestion
When a new version of the nRF54H20 firmware bundle is released, update it as follows:
```

Author: slightly modified

1. Download the new version of the nRF54H20 firmware bundle (:file:`nrf54h20_soc_binaries_v<x.y.z>.zip`).
#. Move the :file:`ZIP` bundle to a folder of your choice and unzip it.
#. |open_terminal_window_with_environment|

> #. Run the following command to verify the current SDFW version::

Reviewer 4: ```suggestion
#. Verify the current SDFW version by running the following command::
```

* When creating a new application specific to your new board, DTS board files can contain all necessary configurations, and no overlay file is needed.
  However, the same limitations regarding the consistency and UICR configuration apply, but should be kept on the board files level.

> * You must manually program the BICR if it has been modified.

Author: This description does not exist, at least not for custom BICRs, just for the one provided for the DK. What would be the process? @hellesvik-nordic @David-Garcia-Polo @tjora @ingeryp
